interop/jvm/src/main/scala/scalaz/zio/interop/catz.scala

```diff
@@ -0,0 +1,65 @@
+package scalaz.zio
+package interop
+
+import cats.effect
+import cats.effect.Effect
+import cats.syntax.all._
+
+import scala.util.control.NonFatal
+
+object catz extends RTS {
```

Comment:
I wonder about extending `RTS` here.... it means users have no control over the thread pool, etc. Maybe that's OK for now. It certainly provides the "easiest" experience, albeit at the cost of more surprises later.

Reply:
I think the other option is to somehow give the instance access to it but that would involve making it implicit maybe, not that sexy either.

BTW I love the design decision of not attaching the ability to evaluate itself to the IO values, I tried to do something really nasty (for debugging purpose in a test), evaluate the IO in a totally wrong place to print its value and I couldn't because of the separation. :smile:

Reply:
This is kinda related to another question i wanted to ask you. How do you plan supporting something like `shift`? do you think it is a valid feature to have? For something like jdbc queries vs http requests, they have different characteristics given one is blocking and the other is not but they might be part of a group of composed IOs which should run in different thread pools. WDYT?

Reply:
`shift` can be implemented now via `async`, it's a few lines of code and I've written it before many times to show people, but refuse to do it again. 😆 

Ultimately composable RTS is the better way to solve it. This will likely involve NOT a `shift`-like design, which is horrible and dysfunctional, but a `forkOn` design.